api/v01/api_base.rb

```diff
@@ -41,11 +41,12 @@ def self.count_locations(obj)
         end
       end
 
-      def self.count_route_locations(params)
+      # Calculate route for given points considering start and end are not the same
+      def self.count_routes(params)
         if params[:loc]
-          count_locations(params[:loc])
+          count_locations(params[:loc]) - 1
         elsif params[:locs]
```

Comment:
It looks like `params[:loc]` is always transformed into `params[:locs]` if this is systematic, do we need this if/elsif logic?

Reply:
You can send a query with `:loc` or `:locs`, it's an inherited thing. For the comparison with nil, `count_locations` returns 0 if no loc is given, I have fixed it.

Reply:
Again out of scope of this PR: I also realized that almost all of the endpoints start with a boiler-plate almost copy-paste like very similar logic that we can uniformise and extract into a single function so that we maintain less code and the  modification are uniform across all endpoints